src/server/pkg/obj/microsoft_client.go

```diff
@@ -34,72 +26,59 @@ var (
 )
 
 type microsoftClient struct {
-	blobClient storage.BlobStorageClient
-	container  string
+	container *storage.Container
 }
 
 func newMicrosoftClient(container string, accountName string, accountKey string) (*microsoftClient, error) {
-	client, err := storage.NewBasicClient(
-		accountName,
-		accountKey,
-	)
+	client, err := storage.NewBasicClient(accountName, accountKey)
 	if err != nil {
 		return nil, err
 	}
-
-	return &microsoftClient{
-		blobClient: client.GetBlobService(),
-		container:  container,
-	}, nil
+	blobSvc := client.GetBlobService()
+	return &microsoftClient{container: (&blobSvc).GetContainerReference(container)}, nil
 }
 
 func (c *microsoftClient) Writer(ctx context.Context, name string) (io.WriteCloser, error) {
-	writer, err := newMicrosoftWriter(ctx, c, name)
-	if err != nil {
-		return nil, err
-	}
-	return newBackoffWriteCloser(ctx, c, writer), nil
+	return newMicrosoftWriter(ctx, c, name), nil
 }
 
 func (c *microsoftClient) Reader(ctx context.Context, name string, offset uint64, size uint64) (io.ReadCloser, error) {
-	byteRange := byteRange(offset, size)
-	var reader io.ReadCloser
-	var err error
-	if byteRange == "" {
-		reader, err = c.blobClient.GetBlob(c.container, name)
-	} else {
-		reader, err = c.blobClient.GetBlobRange(c.container, name, byteRange, nil)
+	blobRange := blobRange(offset, size)
+	if blobRange == nil {
+		return c.container.GetBlobReference(name).Get(nil)
 	}
+	return c.container.GetBlobReference(name).GetRange(&storage.GetBlobRangeOptions{Range: blobRange})
+}
 
-	if err != nil {
-		return nil, err
+func blobRange(offset, size uint64) *storage.BlobRange {
```

Comment:
This is pretty minor, but want to have this return a `*storage.GetBlobRangeOptions`? I think that would make the above code a bit cleaner.